examples/server/server.cpp

```diff
@@ -1117,6 +1120,13 @@ struct server_context {
     }
 
     bool process_token(completion_token_output & result, server_slot & slot) {
+        if (!slot.is_alive()) {
```

Comment:
We're calling `sink.is_writable()` from another thread (other than HTTP thread). Unless `sink` object is thread-safe, I don't think this is a safe way to do.

Reply:
One possible solution could be to introduce thread-safe support into the httplib itself, but I'm not sure how complex it is.

Reply:
Great point, asked on https://github.com/yhirose/cpp-httplib/issues/1952 :-)

Reply:
@yhirose's detailed answer suggests this is *probably* fine. I'd suggest to give it an optimistic try and watch out for crash reports (we could also hide this under a define but the lack of discoverability would reduce feedback opportunities).

Incidentally, I'm keen to explore fuzzing tests at some point, maybe in conjunction w/ k6 / bench (the server would benefit from a bit more hardening)

Reply:
> the server would benefit from a bit more hardening

(just discovered we set CPPHTTPLIB_NO_EXCEPTIONS in debug, maybe some of the crashes I've seen were *fine* haha)

Reply:
As someone who often works with system programming, I think this must be tested very seriously before merging, rather than waiting for bug reports. Thread-safe related stuff is very nasty to debug. Bugs related to thread-safety are usually not consistent, thus many cases are misdiagnosed.

Furthermore, we don't know if `sink.is_writable()` is really atomic on other platforms than linux/unix. Also, because `sink.is_writable()` make syscall under the hood, it potentially stall the inference thread, thus can negatively impact performance.

Reply:
Also want to notice, thread-safety violations **not** always result in crashes.

In the worst case scenario, if a data race happen somewhere, you could end up with some weird behaviors, for example interlaced output (just take this as an example, it can't happen on socket IRL, but same thing may happen)

Reply:
Opened https://github.com/yhirose/cpp-httplib/pull/1956 to maybe introduce an is_alive checker that might be lighterweight as you suggested in the other thread. If it goes through we could use it on platforms where we're happy w/ the threadsafety guarantee 🤞